File: backend/app/schemas/document.py
from datetime import datetime
from pydantic import BaseModel, Field


class DocumentSummary(BaseModel):
    documentId: str
    name: str
    uploadStatus: str  # idle | uploading | uploaded | failed
    processingStage: str  # pending | extracting | comparing | done | error
    validationStatus: str  # pending | structurally_verified | integrity_conflict | training_approved
    version: str
    hash: str
    createdAt: str
    author: str | None = None
    fileSizeBytes: int | None = None
    pageCount: int | None = None
    errorType: str | None = None
    errorMessage: str | None = None


class DocumentVersion(BaseModel):
    documentId: str
    version: str
    name: str
    createdAt: str

File: frontend/src/types/document.ts
export type UploadStatus = 'idle' | 'uploading' | 'uploaded' | 'failed';
export type ProcessingStage = 'pending' | 'extracting' | 'comparing' | 'done' | 'error' | 'cancelled';
export type ValidationStatus = 'pending' | 'structurally_verified' | 'integrity_conflict' | 'training_approved' | 'screenshot_failed' | 'validation_failed';

export interface DocumentSummary {
  documentId: string;
  name: string;
  uploadStatus: UploadStatus;
  processingStage: ProcessingStage;
  validationStatus: ValidationStatus;
  version: string;
  hash: string;
  createdAt: string;
  author?: string;
  fileSizeBytes?: number;
  pageCount?: number;
  errorType?: string;
  errorMessage?: string;
}

export interface DocumentVersion {
  documentId: string;
  version: string;
  name: string;
  createdAt: string;
}
